core/block_drag_surface.js

```diff
@@ -200,11 +200,17 @@ Blockly.BlockDragSurfaceSvg.prototype.getCurrentBlock = function() {
 /**
  * Clear the group and hide the surface; move the blocks off onto the provided
  * element.
- * @param {!Element} newSurface Surface the dragging blocks should be moved to.
- */
-Blockly.BlockDragSurfaceSvg.prototype.clearAndHide = function(newSurface) {
-  // appendChild removes the node from this.dragGroup_
-  newSurface.appendChild(this.getCurrentBlock());
+ * @param {Element} opt_newSurface Surface the dragging blocks should be moved
+ *     to, or null if the blocks should be removed from this surface without
+ *     being moved to a different surface.
+ */
+Blockly.BlockDragSurfaceSvg.prototype.clearAndHide = function(opt_newSurface) {
+  if (opt_newSurface) {
+    // appendChild removes the node from this.dragGroup_
+    opt_newSurface.appendChild(this.getCurrentBlock());
+  } else {
```

Comment:
We hit the else when we're deleting a block.  This is a minor performance optimization: if you drag over a delete area, we don't need to put the block svg back on the workspace svg and *then* remove it from the workspace svg during dispose.  We can just not put it back.